File: php/test/Structural/BridgeTest.php
<?

require_once 'src/Structural/Bridge/SodaImpSingleton.php';
require_once 'src/Structural/Bridge/MediumSoda.php';
require_once 'src/Structural/Bridge/SuperSizeSoda.php';
require_once 'src/Structural/Bridge/CherrySodaImp.php';
require_once 'src/Structural/Bridge/OrangeSodaImp.php';
require_once 'src/Structural/Bridge/GrapeSodaImp.php';

class Structural_BridgeTest extends PHPUnit_Framework_TestCase
{

	function testCherryPlatform() {
		SodaImpSingleton::setTheSodaImp(new CherrySodaImp);
		$mediumSoda = new MediumSoda();
		$this->assertEquals($mediumSoda->pourSoda(), 'Yummy Cherry Soda! Yummy Cherry Soda!', 'Testing medium cherry');
		$superSizeSoda = new SuperSizeSoda();
		$this->assertEquals($superSizeSoda->pourSoda(), 'Yummy Cherry Soda! Yummy Cherry Soda! Yummy Cherry Soda! Yummy Cherry Soda! Yummy Cherry Soda!', 'Testing supersize cherry');
	}
		
	function testGrapePlatform() {
		SodaImpSingleton::setTheSodaImp(new GrapeSodaImp);
		$mediumSoda = new MediumSoda();
		$this->assertEquals($mediumSoda->pourSoda(), 'Yummy Grape Soda! Yummy Grape Soda!', 'Testing medium grape');
		$superSizeSoda = new SuperSizeSoda();
		$this->assertEquals($superSizeSoda->pourSoda(), 'Yummy Grape Soda! Yummy Grape Soda! Yummy Grape Soda! Yummy Grape Soda! Yummy Grape Soda!', 'Testing supersize grape');
	}   

	function testOrangePlatform() {
		SodaImpSingleton::setTheSodaImp(new OrangeSodaImp());
		$mediumSoda = new MediumSoda();
		$this->assertEquals($mediumSoda->pourSoda(), 'Yummy Orange Soda! Yummy Orange Soda!', 'Testing medium orange');
		$superSizeSoda = new SuperSizeSoda();
		$this->assertEquals($superSizeSoda->pourSoda(), 'Yummy Orange Soda! Yummy Orange Soda! Yummy Orange Soda! Yummy Orange Soda! Yummy Orange Soda!', 'Testing supersize orange');
	}
    
   public function test()
   {
		$this->testCherryPlatform();
		$this->testGrapePlatform();
		$this->testOrangePlatform();
	}
}

File: clojure/test/patterns/structural/bridge_test.clj
(ns patterns.structural.bridge-test
  (:require
      [clojure.test :refer :all]
      [clojure.string :as str]
      [patterns.structural.bridge :refer :all])
  (:import (patterns.structural.bridge cherrysodaimp grapesodaimp orangesodaimp mediumsoda supersizesoda)))

(deftest structural-bridge-test
   (let [
		c (->cherrysodaimp)
		g (->grapesodaimp)
		o (->orangesodaimp)
		m (->mediumsoda)
		z (->supersizesoda)
		flavors [c g o]
		sizes [m z]
	]
	(testing "One big munge of flavor and sizes test" 
		(is (= (str
			"...glug...Yummy Cherry Soda!...glug...Yummy Cherry Soda!"
			"...glug...Yummy Cherry Soda!...glug...Yummy Cherry Soda!...glug...Yummy Cherry Soda!...glug...Yummy Cherry Soda!...glug...Yummy Cherry Soda!"
			"...glug...Delicious Grape Soda!...glug...Delicious Grape Soda!"
			"...glug...Delicious Grape Soda!...glug...Delicious Grape Soda!...glug...Delicious Grape Soda!...glug...Delicious Grape Soda!...glug...Delicious Grape Soda!"
			"...glug...Citrusy Orange Soda!...glug...Citrusy Orange Soda!"
			"...glug...Citrusy Orange Soda!...glug...Citrusy Orange Soda!...glug...Citrusy Orange Soda!...glug...Citrusy Orange Soda!...glug...Citrusy Orange Soda!")
		(str/join 
			(map (fn [f] 
				(str/join 
					(map (fn [s](pour s f)) sizes))) 
			flavors))
	)))))


